File: types/category.go
package types

import "time"

type Category struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Slug        string    `json:"slug"`
	Description string    `json:"description"`
	ParentID    *string   `json:"parent_id,omitempty"`
	UpdatedAt   time.Time `json:"updated_at"`
	CreatedAt   time.Time `json:"created_at"`
}

File: web/src/types/category.ts
export interface Category {
  id: string
  name: string
  slug: string
  description?: string
  parent_id?: string
  created_at?: string
  updated_at?: string
}

// FIXME move this to some shared types file
// exists in product.ts too
type RequireOnly<T, K extends keyof T> = Partial<T> & Pick<T, K>

export type UpdateCategoryRequest = RequireOnly<Category, 'id'>
